File: api/src/org/ufomatics/archipelago/model/MetaFieldImpl.java
package org.ufomatics.archipelago.model;

import org.ufomatics.archipelago.model.MetaField;
import org.ufomatics.archipelago.model.MetaType;

import java.io.Serializable;

/**
 * @author <a href="mailto:rr0@rr0.org">Jerome Beau</a>
 * @version $revision$
 */
public class MetaFieldImpl implements MetaField, Serializable {
    private MetaType owner;
    private String name;
    private MetaType type;

    public MetaFieldImpl() {
    }

    public MetaFieldImpl(final String someName, MetaType someType) {
        name = someName;
        type = someType;
    }

    public MetaType getOwner() {
        return owner;
    }

    public MetaType getType() {
        return type;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setType(MetaType type) {
        this.type = type;
    }

    public void setOwner(MetaType owner) {
        this.owner = owner;
    }

    public void retrieve(Object databaseMappings) {
/*
        Iterator iterator = mappings.iterator();
        while (iterator.hasNext()) {
            FieldMapping fieldMapping = (FieldMapping) iterator.next();
            fieldMapping.getDatabase();
        }
*/
    }

    public String toString() {
        return getName() + ": " + type;
    }
}

File: src/archipelago/api/model/MetaFieldImpl.ts
import {MetaField} from "archipelago/api/model/MetaField"
import {MetaType} from "archipelago/api/model/MetaType"
import {JSerializable} from "archipelago/api/util/jsdk/util/JSerializable"

/**
 *
 */
export class MetaFieldImpl implements MetaField, JSerializable {

 // private owner: MetaType

  constructor(protected name: string, protected type: MetaType) {
  }

  /* getOwner(): MetaType {
     return this.owner
   }*/

  getType(): MetaType {
    return this.type
  }

  getName(): string {
    return this.name
  }

  setName(name: string): void {
    this.name = name
  }

  setType(type: MetaType): void {
    this.type = type
  }

  /*setOwner(owner: MetaType): void {
    this.owner = owner
  }*/

  retrieve(_databaseMappings: any): void {
    /*
      Iterator iterator = mappings.iterator();
      while (iterator.hasNext()) {
          FieldMapping fieldMapping = (FieldMapping) iterator.next();
          fieldMapping.getDatabase();
      }
    */
  }

  toString(): string {
    return this.getName() + ": " + this.type
  }
}
